apps/autoscheduler/LoopNest.cpp

```diff
@@ -0,0 +1,1861 @@
+                    outer->size[i] = outer_extent;
+                    const auto &p = parent_bounds->loops(stage->index, i);
+                    int64_t min = p.min();
+                    int64_t original_extent = p.extent();
+                    int64_t inner_extent = (original_extent + outer_extent - 1) / outer_extent;
+                    // Pick a more representative loop iteration
+                    min += (outer_extent / 2) * inner_extent;
+                    bool compile_time_constant_extent =
+                        (p.constant_extent() || outer_extent > 1) &&
+                        (inner_extent == 1 || outer_extent == 1 || stage->index == 0);
+                    b->loops(stage->index, i) = Span(min, min + inner_extent - 1, compile_time_constant_extent);
+                }
+
+                // Region_{computed/required} on outer is now
+                // wrong, but it doesn't matter because consumers
+                // only look at the loops in get_bounds. Still,
+                // this is weird.
+                outer->set_bounds(node, b);
+            }
+
+            if (!in_realization) {
+                outer->store_at.insert(f);
+            }
+            outer->children.emplace_back(inner);
+
+            // HACK
+            // bool may_slide = false;
+            bool may_slide = (!in_realization &&
+                              f->stages.size() == 1);
+            if (may_slide) {
+                // Store here, but compute further in. Currently
+                // don't have to worry about the constraints this
+                // places on parallelism, as we forced all the
+                // parallelism to the outer loop.
+                auto opts = inner->compute_in_tiles(f, outer, params, v, true);
+                for (IntrusivePtr<const LoopNest> &n : opts) {
+                    LoopNest *store_at_outer_compute_further_in = new LoopNest;
+                    store_at_outer_compute_further_in->copy_from(*outer);
+                    store_at_outer_compute_further_in->children.pop_back();
+                    store_at_outer_compute_further_in->children.emplace_back(std::move(n));
+                    result.emplace_back(store_at_outer_compute_further_in);
+                }
+            }
+
+            // Site the computation inside the outer loop
+            outer->compute_here(f, true, v);
+            outer->tileable &= !in_realization;
+            result.emplace_back(outer);
+        }
+    }
+
+    if (child >= 0 && !called_by_multiple_children && !in_realization &&
+        (may_subtile() || is_root())) {
+        // Push the Func further inwards in the loop nest
+
+        // See if it's appropriate to slide over this loop Can't
+        // slide at the root level if we intend to parallelize it.
+        // HACK
+        // bool may_slide = false;
+        bool may_slide = (params.parallelism == 1) || !is_root();
+
+        const auto &c = children[child];
+        int num_ones =
```

Comment:
remove or justify leaving it in (but commented out)

Reply:
I deleted it.